js/src/attribute-mapping/attribute-mapping-rule-modal.js

```diff
@@ -66,23 +99,18 @@ const AttributeMappingRuleModal = ( { rule, onRequestClose = noop } ) => {
 					{ __( 'Target attribute', 'google-listings-and-ads' ) }
 				</Subsection.Title>
 				<Subsection.Subtitle className="gla_attribute_mapping_helper-text">
-					{ __(
-						'Select a Google attribute that you want to manage',
-						'google-listings-and-ads'
-					) }
+					{ attributeSelectorLabel }
 				</Subsection.Subtitle>
 				<AppSelectControl
+					aria-label={ attributeSelectorLabel }
 					onChange={ setSelectedAttribute }
-					options={ attributes.map( ( attribute ) => {
-						return {
-							value: attribute.id,
-							label: attribute.label,
-						};
-					} ) }
+					options={ attributesOptions }
 				/>
 			</Subsection>
 
-			{ sources.length > 0 && (
+			{ ! sourcesHasFinishedResolution && <AppSpinner /> }
+
+			{ sourcesOptions.length > 1 && sourcesHasFinishedResolution && (
```

Comment:
Will the `sourcesOptions.length` always be at least 2? Or is it possible the fetched attribute source array has only one item?

Reply:
I'm counting the empty one as well "Select Option, coming from backend" 

Note: This is maybe related with the fix on the sources so I need to equalise a bit the response format in backend (including in both empty option and also same value/label format) 

Thanks!

Reply:
Sorry, I said this comes (or should come from backend) but it's in `<AttributeMappingFieldSourcesControl>`

So you were right. its ` length > 0`

Reply:
Thanks for clarifying!